geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureConverterFactory.scala

```diff
@@ -257,6 +272,7 @@ trait ToSimpleFeatureConverter[I] extends SimpleFeatureConverter[I] with LazyLog
   def inputFields: Seq[Field]
   def idBuilder: Expr
   def userDataBuilder: Map[String, Expr]
+  def caches: Map[String, EnrichmentCache]
```

Comment:
maybe we should provide a default implementation here of `= Map.empty` to not break other `ToSimpleFeatureConverter` impls, or do we not care/want to force them to implement the caching?

Reply:
Hmm, we could provide a default impl but then if a custom impl went to use the caches, they would have to hunt this down.  I could see it going both ways.